### ISSUE TITLE
router with default route to page with child router results in invalid routes

### ISSUE THREAD

**Issue by mbroadst** [aurelia/router#117]
This plunker demonstrates the issues: http://plnkr.co/edit/06iAL0HKrmWtofsTFehZ

On the main page, there are two routes (toptestone, and toptesttwo - excuse the names I was trying to rule out casing as an issue). "toptestone" is the default route, and the module it routes too has a child router in it. When one tries to navigate to the *non* default route of the child router page, it results in an inability to find the route. I think the only reason this works on the skeleton-navigation is because the child router there is using the exact same router configuration as the top level.

TL;DR - load the plunker above, and click on "Sub Test 2"

**Comment by mbroadst** [comment]
To help clear up some issues with naming, I've renamed everything to People, Animal, Squirrel and Rabbit :smile: over here http://plnkr.co/edit/VHGKRin7Do2IMgL6hf7n

**Comment by mbroadst** [comment]
Is it possible to force aurelia to always inject the route, even when using the default route? This would fix the problem entirely for me. For instance, if I hade a top level route ['', 'welcome'], and navigated to my.site.com/, and it rewrote the address as my.site.com/welcome

**Comment by EisenbergEffect** [comment]
@bryanrsmith Can you look into this. I think we have a duplicate of this somewhere else as well.

**Comment by jadrake75** [comment]
Just to clarify - you are wanting the default "" to be map to welcome on the address bar?  what if you had something like [ '', 'welcome', 'welcome-error'] ?  what would it map to?

**Comment by mbroadst** [comment]
@jadrake75 that was a suggestion for a stopgap solution for the particular case I ran into. I'm not exactly sure what ['', 'welcome', 'welcome-error'] would route to today? Do you have some sort of logic to route to those based on some flags?

The real problem is illustrated by the plunker posted above, and specifically is that if a default route is followed to a page with a child router, none of those sub routes work because the router doesn't "remember" that its now a level deep into the routes because it's replaced the route with '' and therefore sub-routes don't append the correct initial path - so that needs to be solved somehow

**Comment by mbroadst** [comment]
also for anyone else checking this out, I was able to workaround the issue by providing a redirect for the default route (only on the top level! if you try this trick on the sublevel you'll run into infinite redirect loops and totally bork your browser):

```
    config.map([
      { route: ['', 'welcome'], title: 'Welcome', moduleId: './welcome', nav: true }
    ]);
```

changes to:
```
    config.map([
      { route: '', redirect: 'welcome' },
      { route: ['welcome'], title: 'Welcome', moduleId: './welcome', nav: true }
    ]);
```

**Comment by EisenbergEffect** [comment]
Our next release fixes redirects on child routers :)

**Comment by mbroadst** [comment]
Okay that works for me, do you guys want me to close this issue? Or would be interesting in the future to sort of "complete" the feature to support the example in the plunker?

**Comment by bryanrsmith** [comment]
The original issue is a duplicate of #27. I'm going to go ahead and close this one.
